roles/pulp_database_config/tasks/main.yml

```diff
@@ -20,23 +20,28 @@
 - meta: flush_handlers
 
 - block:
-    - name: Run database auth migrations
-      command: '{{ pulp_django_admin_path }} migrate auth --no-input'
-      register: migrate_auth
-      changed_when: "'No migrations to apply' not in migrate_auth.stdout"
-      environment:
-        # Stops django-guardian from creating an anonymous user
-        PULP_ANONYMOUS_USER_NAME: '@none None'
-
     - name: Run database migrations
       command: '{{ pulp_django_admin_path }} migrate --no-input'
       register: result
       changed_when: "'No migrations to apply' not in result.stdout"
+      environment:
+        # Stops django-guardian from creating an anonymous user
+        PULP_ANONYMOUS_USER_NAME: '@none None'
+
+    - name: Check if admin password has been set
+      stat:
+        path: "{{ pulp_user_home }}/admin_password_is_set"
```

Comment:
Users that used older versions will have their password reset one last time here. Not so sure how to solve that.
Maybe tell them to touch that file.

Reply:
Yeah, I planned to research how to check whether it has been set as part of issue #7499.

Or just use some other way of determining whether Pulp 3 was installed in the past or not.

Reply:
To clarify, I hope we can determine feasibly whether it was set or not. I really want us to do it that way. Partially to address existing installs, and partially because I hate cluttering up the filesystem.

If it isn't feasible, then this comment can be overcome.